[docs] Add HTML to languages-and-frameworks.md (#386)
gitpod-io/website: src/docs/html_in_gitpod.md

# HTML

> Looking to try web-development in Gitpod? Well your in luck Gitpod's IDE - Theia - was made with web-development in mind, but depending on your needs you may want to customize this experience further!

Reviewer: Typos:

```suggestion
Looking to try web development in Gitpod? Well you're in luck: Gitpod was made with web development in mind. And, depending on your needs, you may want to customize this experience further!
```

## Workspace Configuration

### VSCode Extensions

#### [HTML-Snippets](https://marketplace.visualstudio.com/items?itemName=abusaidm.html-snippets)

> This extension adds many usful snippets for productivity, see for yourself.

Reviewer: Typos:

```suggestion
This extension adds many useful snippets for productivity. See for yourself:
```

![Example](images/HTML-Snippets-Example.gif)
#### [HTML CSS Support](https://marketplace.visualstudio.com/items?itemName=ecmel.vscode-html-css)

> This will add great auto-complete for your project including color comlpetion!

Reviewer: Typos:

```suggestion
This will add great auto-completion for your project, including color-completion!
```

![HTML-CSS-Support](images/HTML_CSS_Extension.png)

> ### Pre-Builds

Reviewer: As a developer who is new to Gitpod, I don't think this section title gives me any useful information.

Maybe `## Configuration files` instead? Or directly `## .gitpod.yml`?

> With Gitpod you can have your project built before you open your repository, no more waiting for all of those Node/Yarn dependencies to be built! Let's get started with building your project to do this we need a [`.gitpod.yml`](https://www.gitpod.io/docs/41_config_gitpod_file/) file. Here is an example of what your start tasks should look like for a typical NPM project

Reviewer: I suggest using more punctuation:

```suggestion
With Gitpod you can have your project built before you open your repository, no more waiting for all of those Node/Yarn dependencies to be built! Let's get started with building your project: To do this we need a [`.gitpod.yml`](https://www.gitpod.io/docs/41_config_gitpod_file/) file. Here is an example of what your start tasks should look like for a typical NPM project:
```

```YAML
tasks:
  - init: npm install
    command: npm start
```

For Yarn
```YAML
tasks:
  - init: yarn && yarn build
    command: yarn run
```

> These should prebuild your workspace whenever you push a change!

Reviewer: That's not exactly true. By default, this will only build your workspace whenever you create a new one.

To build things _before_ a workspace is opened (i.e. prebuilds), you need to install the [Gitpod app](https://github.com/marketplace/gitpod-io/) first. You'll also need to make sure that it's enabled for your workspace.

Additionally, keep in mind that only the `init:` instructions are prebuilt. The `command:` is supposed to be a quick command that starts an app interactively, which is why is doesn't fit well with prebuilds conceptually.

> Next lets talk about previewing your setup without leaving the editor. To do this we must configure ports say your server listens on port 3000 in your [`.gitpod.yml`](https://www.gitpod.io/docs/41_config_gitpod_file/) file add this

Reviewer: Typos:

```suggestion
Next let's talk about previewing your HTML pages without leaving the editor. To do this we must configure ports say your server listens on port 3000 in your [`.gitpod.yml`](https://www.gitpod.io/docs/41_config_gitpod_file/) file add this
```

Reviewer: I think there are several possible scenarios here, when you want to preview HTML files.

1) You want to live-preview just a single HTML file. For this, you can click on the small Live Preview icon on the top-right corner of your HTML editor, and this will open an HTML live preview.

2) You want to preview several files, e.g. an HTML file that imports external scripts, CSS files, images, etc. For this, you can simply:
  a) run `python3 -m http.server 3000` in your workspace directory
  b) use the Gitpod notification to open port 3000 in a Preview
  c) open the HTML file you want to preview

3) You already have a server (e.g. via NPM or Yarn or something else like Ruby-on-Rails etc). Then you can configure your server port to automatically open a Preview like you explained.

```YAML
# Declare ports section
ports:
    # Expose port 3000
  - port: 3000
    # Open preview after port opens
    onOpen: open-preview
```
### Try It!

> Want to try it out? And see a minimal example in action? Great, then open it in Gitpod!

Reviewer: Maybe a little bit less insistent on the usage of Gitpod? 😅 

```suggestion
Want to try it out? And see a minimal example in action? Great, then you can open this in Gitpod:
```

Author: Sorry my bad!

[![JesterOrNot/Gitpod-Web-Development-Example](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/JesterOrNot/Gitpod-Web-Development-Example)
## External Resources
* [Start Tasks](https://www.gitpod.io/docs/44_config_start_tasks/)
* [Exposing Ports](https://www.gitpod.io/docs/43_config_ports/)
* [Theia](https://www.gitpod.io/docs/50_ide/)

> * [Prebuilds](https://www.gitpod.io/docs/46_prebuilds/)

Reviewer: These external links feel a bit random. But I'm not opposed to having them.

Author: Added descriptions
